docs/entities/clusterenvironment.md

```diff
@@ -78,13 +86,17 @@ status:
 
 ### Creation
 
-When a Cluster Environment is created, Primaza must verify the connection and update the field `status.state` accordingly.
+When a Cluster Environment is created, Primaza verifies the connection to the cluster and, if the connection test passes, its permissions in application and service namespaces.
+For each service and application namespace on which permissions are granted, Primaza pushes respectively the service or application agent.
+
+Cluster Environment's State and Conditions are updated according to tests and agents' deployment results.
 
 ### Deletion
 
-No operations should be performed.
+When a Cluster Environment is deleted, the permissions granted in Primaza's namespace to Users associated to namespace agents and agent deployments on target cluster's namespaces are removed.
 
 ### Update
 
-When a Cluster Environment is updated, Primaza must verify the connection and update the field `status.state` accordingly.
+As on [creation](#creation), Primaza verifies the connection and its permissions and reconciles application and service namespaces.
```

Comment:
Something like this: As on [creation](#creation), Primaza verifies the connection and permissions to application and service namespaces.

Reply:
What do you think about the following?

```suggestion
As on [creation](#creation), Primaza verifies the connection to and its permissions into the target cluster. Finally, it pushes agents in cluster's application and service namespaces.
```